deckhouse-controller/pkg/controller/module-controllers/release/override-controller.go

```diff
@@ -211,14 +218,24 @@ func (c *modulePullOverrideReconciler) moduleOverrideReconcile(ctx context.Conte
 		return ctrl.Result{RequeueAfter: mo.Spec.ScanInterval.Duration}, fmt.Errorf("got an empty module definition for %s module pull override", mo.Name)
 	}
 
-	err = validateModule(*moduleDef)
+	var values = make(addonutils.Values)
+	if module := c.moduleManager.GetModule(moduleDef.Name); module != nil {
+		values = module.GetConfigValues(false)
+	}
+	err = validateModule(*moduleDef, values)
 	if err != nil {
 		mo.Status.Message = fmt.Sprintf("validation failed: %s", err)
 		if e := c.updateModulePullOverrideStatus(ctx, mo); e != nil {
-			return ctrl.Result{Requeue: true}, e
+			return ctrl.Result{Requeue: true}, fmt.Errorf("update override status: %w", e)
 		}
+		return ctrl.Result{}, fmt.Errorf("validation failed: %w", err)
+	}
 
-		return ctrl.Result{RequeueAfter: mo.Spec.ScanInterval.Duration}, nil
+	if err = os.RemoveAll(c.downloadedModulesDir); err != nil {
```

Comment:
looks suspiciously :thinking:

Reply:
Similar to the comment above,`(md *ModuleDownloader) storeModule` is also used with directory cleaning before copying. And because of the temporary directory, we need to do this ourselves.

Reply:
c.downloadedModulesDir is `/var/lib/deckhouse/downloaded` [directory](https://github.com/deckhouse/deckhouse/pull/9293/files/7279fc35d747bcddf9f1ee10b54c415237a895cd#diff-4fcd64acf898a13f2e50768163269ec98a0c0cd0f93bef40fa24515d107bed8cL76), I definitely don't want to have it cleaned every time an MPO is downloaded.

Reply:
At the moment this is the implementation that already works. Do you want to change this behavior?

Reply:
I apologize, it took me a long time to figure out what I did wrong. Today I figured it out. Thank you very much for noticing such a gross mistake on my part.

I fixed it and tested it on the dev cluster. Now validation in case of using mpo goes as expected.